src/lazy/encoder/writer.rs

```diff
@@ -207,9 +228,10 @@ impl<'value, V: ValueWriter> AnnotatableWriter for ApplicationValueWriter<'value
     where
         Self: 'a,
     {
-        if self.encoding.symbol_creation_policy == SymbolCreationPolicy::WriteProvidedToken {
+        if self.has_inline_symbol_text() {
             // Store the tokens as they are. Text will be written as text, symbol IDs will be written
-            // as symbol IDs. TODO: Lookup SIDs to see if they have text?
+            // as symbol IDs.
+            // TODO: Replace SIDs w/text if they're defined? (Config setting?)
```

Comment:
🗺️ If the caller passes a SID, we write it as-is. At a minimum, we should validate that it's in range.